scripts/eachdist.py

```diff
@@ -503,6 +515,117 @@ def lint_args(args):
     )
 
 
+def update_changelog(path, version, new_entry):
+    unreleased_changes = 0
+    try:
+        with open(path) as changelog:
+            text = changelog.read()
+            if version in text:
+                raise AttributeError(
+                    "{} already contans version {}".format(path, version)
+                )
+        with open(path) as changelog:
+            for line in changelog:
+                if line.startswith("## Unreleased"):
+                    unreleased_changes = 0
+                elif line.startswith("## "):
+                    break
+                elif len(line.strip()) > 0:
+                    unreleased_changes += 1
+
+    except FileNotFoundError:
+        print("file missing: {}".format(path))
+        return
+
+    if unreleased_changes > 0:
+        print("updating: {}".format(path))
+        text = re.sub("## Unreleased", new_entry, text)
+        with open(path, "w") as changelog:
+            changelog.write(text)
+
+
+def update_changelogs(targets, version):
+    print("updating CHANGELOG")
+    today = datetime.now().strftime("%Y-%m-%d")
+    new_entry = "## Unreleased\n\n## Version {}\n\nReleased {}".format(
+        version, today
+    )
+    errors = 0
+    for target in targets:
+        try:
+            update_changelog(
+                "{}/CHANGELOG.md".format(target), version, new_entry
+            )
+        except Exception as e:
+            print(str(e))
+            errors += 1
+
+    if errors != 0:
+        sys.exit(1)
+
+
+def find(name, path):
+    for root, dirs, files in os.walk(path):
+        if name in files:
+            return os.path.join(root, name)
+
+
+def update_version_files(targets, version):
+    print("updating version.py files")
+    update_files(
+        targets,
+        version,
+        "version.py",
+        "__version__ .*",
+        '__version__ = "{}"'.format(version),
+    )
+
+
+def update_dependencies(targets, version):
+    print("updating dependencies")
+    update_files(
+        targets,
+        version,
+        "setup.cfg",
+        r"(opentelemetry-.*)= (.*)",
+        r"\1= " + version,
+    )
+
+
+def update_files(targets, version, filename, search, replace):
+    errors = 0
+    for target in targets:
+        curr_file = find(filename, target)
+        if curr_file is None:
+            print("file missing: {}/{}".format(target, filename))
+            continue
+
+        with open(curr_file) as f:
+            text = f.read()
+
+        if version in text:
+            print("{} already contans version {}".format(curr_file, version))
+            errors += 1
+            continue
+
+        with open(curr_file, "w") as f:
+            f.write(re.sub(search, replace, text))
```

Comment:
This could be a problem if an error is found later, this would leave some files updated and others don't. It would be better to only write files after a run has been made that did not find any errors, to keep changes atomic.

Reply:
Since this code doesn't actually commit the files, there's not a huge advantage one way or the other.